### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by anastasiiashchoholieva**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,24 +2,78 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, instance: None, name: str) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py:9]: add typehints
https://stackoverflow.com/questions/3798835/understanding-get-and-set-and-python-descriptors

File: app/main.py [hunk 2]
```diff
@@ -2,24 +2,78 @@
...
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: None, objtype: None) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py:12]: typehints are invalid
https://stackoverflow.com/questions/3798835/understanding-get-and-set-and-python-descriptors

File: app/main.py [hunk 3]
```diff
@@ -2,24 +2,78 @@
...
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: None, value: int) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py:15]: typehints are invalid
https://stackoverflow.com/questions/3798835/understanding-get-and-set-and-python-descriptors

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: app/main.py [hunk 4]
```diff
@@ -1,25 +1,80 @@
...
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: IntegerRange) -> None:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError
```
Comment (ArsenPidhoretskyi) [inline app/main.py:20]: Add messages for error

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1,25 +1,80 @@
...
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(
+        self,
+        name: str,
+        limitation_class: SlideLimitationValidator
```
Comment (ArsenPidhoretskyi) [inline app/main.py:66]: it does not accept instance of SlideLimitationValidator it accept type[SlideLimitationValidator]

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by olena-hul — APPROVED** [review]

**Comment by olena-hul** [comment]
Good job!

**Pull request closed without merging**